### PR TITLE
Use RCTEventDispatcherProtocol in RCTView

### PR DESCRIPTION
**Pull request by christophpurrer**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary
This change makes the macOS fork compatible with react-native-core and the current RTCBridge implementation.
It changed in react-native-core in https://github.com/facebook/react-native/commit/ea93151f21003df6f65dd173dd5dcb3135b0ae94

## Changelog

[macOS] [Fixed] - Use RCTEventDispatcherProtocol in RCTView.h/.m (instead of RCTEventDispatcher*)

## Test Plan

Ran rn-tester app

Screen Shot 2022-02-21 at 3 16 51 PM

Used at Meta Inc. for Facebook Messenger macOS

### CODE REVIEW

**Comment by Saadnajmi** [comment]
Thanks for the PR! Just to double check, was `RCTEventDispatcherProtocol` introduced in RN 0.64? I'm basing that off the commit you linked and seeing what tags are present. We're currently synced up to approximately RN 0.66.4, so I wanted to make sure that `RCTEventDispatcherProtocol` wasn't introduced in say, 0.67

**Comment by christophpurrer** [comment]
Hi @Saadnajmi,
thanks for the follow up. So by looking at the Git history, Meta/Facebook made this change in
0.64-stable > https://github.com/facebook/react-native/blob/0.64-stable/Libraries/NativeAnimation/RCTNativeAnimatedModule.h
It is also in 0.64.0 https://github.com/facebook/react-native/blob/v0.64.0/Libraries/NativeAnimation/RCTNativeAnimatedModule.h

We (Meta) don't use 0.64 macOS - but we try to upgrade to 0.66 and this change here fixes a compile error.
Shall we 'just' pick the change into the 0.66-stable branch?

File: Libraries/Components/Pressable/Pressable.js
Comment (Saadnajmi) [inline Libraries/Components/Pressable/Pressable.js:35]: These changes in `Pressable` look like a difference between `main` and `0.66-stable`, and not part of your change? They should go away if you target main again, but also tell me I need to go update 0.66-stable with these changes.

Comment (christophpurrer) [reply Libraries/Components/Pressable/Pressable.js]: Afaik the main branch should be (at least) rn-0.66, so lets target the main branch again ;-)
I am curious why this did not create an issue already, don't you have subclasses of RCTView?

Comment (Saadnajmi) [reply Libraries/Components/Pressable/Pressable.js]: Yeah, both `main` and `0.66-stable` are about equal at the moment. 
It looks like the RCTEventDispatcher/Protocol mismatch has been causing a warning, but not an error. I wonder why we also have it, given RN Core doesn't have an `initWithEventDispatcher` on `RCTView`. My guess is it's because our `RCTView` has focus/blur events..

**Review by Saadnajmi — APPROVED** [review]
The change looks good to me! I'll leave it to @heyimchris to take a second look / merge in if it looks good!

**Comment by christophpurrer** [comment]
@Saadnajmi  > the build failed due to 

error: Build input file cannot be found: '/Users/runner/work/1/s/react-native-test-app/lib/clang/13.0.0/lib/darwin/libclang_rt.ubsan_osx_dynamic.dylib' (in target 'ReactTestApp' from project 'ReactTestApp')


I guess that's unrelated to my change ;-)
How can we fix the build error?

**Comment by Saadnajmi** [comment]
> @Saadnajmi > the build failed due to
> 
> error: Build input file cannot be found: '/Users/runner/work/1/s/react-native-test-app/lib/clang/13.0.0/lib/darwin/libclang_rt.ubsan_osx_dynamic.dylib' (in target 'ReactTestApp' from project 'ReactTestApp')
> 
> I guess that's unrelated to my change ;-) How can we fix the build error?

Hmm, I think this might be because React Native macOS builds with ASAN on Debug, and clang static analysis flags on ship, which React-Native-Test-App (what the Integration CI is testing against) builds with ASAN on both Debug and Ship. That mismatch might cause the issue. I'll defer to @heyimchris and @tido64 who enabled ASAN in each repo. That CI check is optional on main anyway, so it won't block merging your PR.

**Comment by tido64** [comment]
> > @Saadnajmi > the build failed due to
> > error: Build input file cannot be found: '/Users/runner/work/1/s/react-native-test-app/lib/clang/13.0.0/lib/darwin/libclang_rt.ubsan_osx_dynamic.dylib' (in target 'ReactTestApp' from project 'ReactTestApp')
> > I guess that's unrelated to my change ;-) How can we fix the build error?
> 
> Hmm, I think this might be because React Native macOS builds with ASAN on Debug, and clang static analysis flags on ship, which React-Native-Test-App (what the Integration CI is testing against) builds with ASAN on both Debug and Ship. That mismatch might cause the issue. I'll defer to @HeyImChris and @tido64 who enabled ASAN in each repo. That CI check is optional on main anyway, so it won't block merging your PR.

This is Xcode's super naïve way of finding `libclang_rt.ubsan_osx_dynamic.dylib` (basically `dirname $(which clang)` and concatenate a hard-coded path). We're hitting this now because RNTA integrated Ccache to speed up building on CI. I'll add a way to disable it.

Regardless, as @Saadnajmi mentioned, it's not a required check so it won't block this PR (even though it's kinda annoying to have a red X there).

**Review by HeyImChris — APPROVED** [review]

**Pull request merged**
